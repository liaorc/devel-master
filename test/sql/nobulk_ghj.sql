--#
--#       Gpu Hash Join TestCases without BulkLoad 
--#

set pg_strom.gpu_setup_cost=0;
set pg_strom.enable_gpupreagg to off;
set pg_strom.enable_gpusort to off;
set random_page_cost=1000000;   --# force off index_scan.                                                                                                                                          
set client_min_messages to warning;

--smallint
select a.id,case when a.smlint_x = b.smlint_x then a.smlint_x else 0 end from 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as a
 inner join 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as b
 on a.smlint_x=b.smlint_x and a.id=b.id order by a.id;

--integer
select a.id,case when a.integer_x = b.integer_x then a.integer_x else 0 end from 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as a
 inner join 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as b
 on a.integer_x=b.integer_x and a.id=b.id order by a.id;

--bigint
select a.id,case when a.bigint_x = b.bigint_x then a.bigint_x else 0 end from 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as a
 inner join 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as b
 on a.bigint_x=b.bigint_x and a.id=b.id order by a.id;

--real
select a.id,case when a.real_x = b.real_x then a.real_x else 0 end from 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as a
 inner join 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as b
 on a.real_x=b.real_x and a.id=b.id order by a.id;

--float
select a.id,case when a.float_x = b.float_x then a.float_x else 0 end from 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as a
 inner join 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as b
 on a.float_x=b.float_x and a.id=b.id order by a.id;

--numeric
select a.id,case when a.nume_x = b.nume_x then a.nume_x else 0 end from 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as a
 inner join 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as b
 on a.nume_x=b.nume_x and a.id=b.id order by a.id;

--small serial
select a.id,case when a.smlsrl_x = b.smlsrl_x then a.smlsrl_x else 0 end from 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as a
 inner join 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as b
 on a.smlsrl_x=b.smlsrl_x and a.id=b.id order by a.id;

--serial
select a.id,case when a.serial_x = b.serial_x then a.serial_x else 0 end from 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as a
 inner join 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as b
 on a.serial_x=b.serial_x and a.id=b.id order by a.id;

--big serial
select a.id,case when a.bigsrl_x = b.bigsrl_x then a.bigsrl_x else 0 end from 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as a
 inner join 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as b
 on a.bigsrl_x=b.bigsrl_x and a.id=b.id order by a.id;
