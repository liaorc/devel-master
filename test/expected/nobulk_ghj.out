--#
--#       Gpu Hash Join TestCases without BulkLoad 
--#
set pg_strom.gpu_setup_cost=0;
set pg_strom.enable_gpupreagg to off;
set pg_strom.enable_gpusort to off;
set random_page_cost=1000000;   --# force off index_scan.                                                                                                                                          
set client_min_messages to warning;
--smallint
select a.id,case when a.smlint_x = b.smlint_x then a.smlint_x else 0 end from 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as a
 inner join 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as b
 on a.smlint_x=b.smlint_x and a.id=b.id order by a.id;
  id   | case 
-------+------
 20100 |  -13
 20200 |   -9
 20300 |   30
 20400 |   -8
 20500 |  -27
 20700 |   -3
 20800 |    6
 20900 |    2
 21000 |   21
 21100 |  -16
 21200 |   25
 21300 |   22
 21400 |   22
 21500 |  -19
 21600 |  -17
 21700 |  -19
 21800 |    3
 21900 |   33
 22000 |  -27
 22100 |   29
 22200 |  -17
 22300 |  -10
 22400 |  -30
 22500 |  -19
 22600 |  -21
 22700 |  -30
 22800 |  -16
 22900 |  -26
 23000 |  -14
 23200 |   29
 23300 |  -11
 23400 |   17
 23500 |  -30
 23600 |  -19
 23700 |   26
 23800 |  -25
 23900 |   -6
 24000 |  -16
 24100 |  -28
 24200 |   -5
 24300 |  -27
 24400 |  -16
 24500 |   -7
 24600 |   -4
 24700 |   25
 24800 |  -21
 24900 |   -5
 25000 |    3
 25200 |  -32
 25300 |  -24
 25400 |   -1
 25500 |   30
 25600 |   -3
 25700 |  -12
 25800 |   28
 25900 |  -29
 26000 |   32
 26100 |   23
 26200 |   27
 26300 |   15
 26400 |    4
 26600 |  -13
 26700 |  -18
 26800 |  -26
 26900 |   29
 27000 |  -16
 27100 |  -10
 27200 |  -17
 27300 |  -14
 27400 |   -3
 27500 |  -30
 27600 |    4
 27700 |  -22
 27800 |  -23
 27900 |   16
 28000 |  -18
 28100 |   16
 28200 |    4
 28300 |   12
 28400 |  -27
 28500 |  -26
 28600 |   11
 28700 |    8
 28800 |  -12
 28900 |   31
 29000 |   19
 29100 |  -14
 29200 |    7
 29300 |   15
 29400 |   17
 29500 |  -30
 29600 |   -7
 29800 |   22
 30000 |    8
(94 rows)

--integer
select a.id,case when a.integer_x = b.integer_x then a.integer_x else 0 end from 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as a
 inner join 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as b
 on a.integer_x=b.integer_x and a.id=b.id order by a.id;
  id   |   case   
-------+----------
 20100 |   177423
 20200 |   676170
 20300 |  -178979
 20400 | -1998734
 20500 |   922198
 20600 | -1834221
 20700 | -1732233
 20800 | -2140602
 20900 |  1567586
 21000 |  -119758
 21100 | -2081552
 21200 |  1797463
 21300 |   997310
 21500 |  1633749
 21600 |  1703319
 21700 | -1013406
 21800 | -1988767
 21900 | -1457854
 22000 |  1711855
 22100 |  1701063
 22200 |  2037450
 22300 |   -58454
 22400 |  -410127
 22500 |  -942645
 22600 | -1812794
 22700 |  1936225
 22800 |  -316384
 22900 |   878476
 23000 | -2002031
 23100 |   263485
 23200 | -1571158
 23300 |   203953
 23400 |   368386
 23500 |   110248
 23600 |   855907
 23700 |   281936
 23800 | -1497526
 23900 |  -399850
 24000 | -1847493
 24100 | -1932290
 24200 |  1875904
 24300 |  1311815
 24500 |    41285
 24600 | -1140932
 24700 | -1024682
 24800 |   273507
 24900 |  -795105
 25100 |   877220
 25200 |  1338764
 25300 |   517826
 25400 |  1343694
 25500 |  1998626
 25600 |    -1553
 25700 | -1518441
 25800 | -1545386
 25900 | -1260720
 26100 |  -139442
 26200 |  -664617
 26300 |  1250175
 26400 |  -946393
 26500 |  1666873
 26600 | -1873230
 26800 |   574288
 26900 |   146395
 27000 |   754406
 27100 |   341813
 27200 |   374638
 27300 |   857821
 27400 |   300321
 27500 |  1943735
 27600 |   209067
 27700 |  -419855
 27800 |  1498361
 27900 | -1473639
 28000 |   307038
 28100 |  1752987
 28200 |   904500
 28300 |  -661884
 28400 |  -783509
 28500 |   424340
 28600 | -1645276
 28700 |   437862
 28800 |  1218068
 28900 |   576871
 29100 |  1983992
 29200 |  -940103
 29300 | -1430247
 29500 |  1582030
 29600 | -1088448
 29700 |   500375
 29800 |  1985563
 29900 |  -188119
 30000 |   800891
(93 rows)

--bigint
select a.id,case when a.bigint_x = b.bigint_x then a.bigint_x else 0 end from 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as a
 inner join 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as b
 on a.bigint_x=b.bigint_x and a.id=b.id order by a.id;
  id   |       case        
-------+-------------------
 20100 |  6205740779327980
 20200 |  3740979358777475
 20300 | -2708091959481729
 20400 |  5170635175452738
 20500 | -3522664942761148
 20600 |   497547522877686
 20700 |  1103945777782194
 20800 | -1890293636151640
 20900 |  3905506164179206
 21000 |  -676321533664166
 21100 | -7479301799246037
 21200 | -4145441316714578
 21300 |  6965783672406409
 21400 |  5446779505043571
 21500 |  2082228740993057
 21600 | -2718684723223200
 21700 |  5145890047794348
 21800 |  7866529616792388
 21900 | -3119629506671804
 22000 |  7944816897684406
 22100 |  8778598838727344
 22200 |  5658058111794020
 22300 | -7378367782867436
 22400 | -5533208978776916
 22500 | -3156565340654142
 22600 | -6914643943489536
 22700 |  4352604984018206
 22800 | -8779678997232484
 22900 |  -165055438566457
 23000 |   187879762360795
 23100 | -5905750585151324
 23200 | -6362805027513303
 23300 | -4016772695050944
 23400 |  5434773353464332
 23500 | -2961946783195857
 23600 | -6560348723988136
 23700 |  5052471837636690
 23800 |  5344877265597497
 23900 |  6916096338220286
 24000 | -6669085295784231
 24100 | -5791544527122596
 24200 | -2824648094765810
 24300 |  4526675438679884
 24400 | -9049450952565720
 24600 | -1953919702581379
 24700 |  5471471349337686
 24800 |  3912363534733804
 24900 | -6372161270170255
 25000 |   550322955729502
 25100 | -5460266999433134
 25200 |  1492970549515649
 25300 | -2099492937564946
 25400 | -3773489941925855
 25500 | -1796459441231495
 25600 | -7061188047240430
 25700 | -7967589183654986
 25800 | -1017333596140077
 25900 | -5091282562282684
 26000 | -1049891415338779
 26100 |   893156427936301
 26200 | -3536550597798920
 26300 | -3482467235736322
 26400 |  6864375233940292
 26500 | -7990150243133948
 26600 |  2775408012704088
 26700 |  1391113269787230
 26800 |  6828208130384462
 26900 | -6241373803700552
 27000 |  7288679245000212
 27100 |  2820054601472934
 27200 |  8101327438685929
 27300 |  6189557754873512
 27400 |   493033065572991
 27500 |  4794758587102527
 27600 |  7121343213159318
 27700 |  6490376448241566
 27800 |   774679660586860
 27900 | -1869858430865375
 28000 | -1928605886893261
 28100 | -6189261032762900
 28200 |  8578503545131631
 28300 |  6880533304634769
 28400 |  4841361043848430
 28500 |  3936578835226558
 28600 |  8630336378690863
 28700 | -7862983949490848
 28800 |  2375531516471018
 28900 |  4960208277896954
 29000 | -2271442838023570
 29100 |   875763708763046
 29200 | -1322621185799750
 29300 |  9084215431471824
 29400 | -3830265518338605
 29500 | -1918337659470610
 29600 | -5099055533965443
 29700 | -7164862098135056
 29800 | -2796074957245776
 29900 |  8499332067912319
 30000 | -2498384400228548
(99 rows)

--real
select a.id,case when a.real_x = b.real_x then a.real_x else 0 end from 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as a
 inner join 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as b
 on a.real_x=b.real_x and a.id=b.id order by a.id;
  id   |  case   
-------+---------
 20100 | -0.4474
 20200 | -0.1755
 20300 | -0.3392
 20400 |  0.3141
 20500 | -0.3896
 20600 |  0.5117
 20700 | -0.1882
 20800 | -0.3214
 20900 |  0.8646
 21000 | -0.6153
 21100 | -0.8229
 21200 |  0.7615
 21300 | -0.5443
 21400 | -0.9983
 21500 |    0.67
 21600 | -0.7606
 21800 | -0.3592
 21900 |   0.483
 22000 | -0.0223
 22100 |   0.625
 22200 |  0.9292
 22300 |  0.0659
 22400 |  0.1658
 22500 |  -0.367
 22600 |   0.886
 22700 | -0.8372
 22800 |  0.6776
 22900 | -0.5315
 23000 |  -0.456
 23100 | -0.7307
 23300 | -0.0188
 23400 |  0.9651
 23500 | -0.6613
 23600 |  0.7465
 23700 | -0.2111
 23800 | -0.0942
 23900 | -0.7299
 24100 |   0.735
 24200 |  0.5369
 24300 | -0.7496
 24400 |  -0.158
 24600 | -0.7266
 24700 | -0.6547
 24900 |  0.9453
 25000 | -0.2331
 25100 |  0.6414
 25200 | -0.6868
 25300 | -0.0803
 25400 | -0.9669
 25500 |  0.1526
 25600 | -0.8438
 25700 |  0.4433
 25800 | -0.0515
 25900 | -0.1171
 26000 |  0.0278
 26100 |   0.143
 26200 | -0.8803
 26300 |  -0.302
 26400 | -0.7165
 26500 |  0.0569
 26600 |  0.5957
 26700 | -0.6223
 26800 | -0.8399
 26900 | -0.7746
 27000 | -0.7199
 27100 | -0.6918
 27200 |  0.5069
 27300 | -0.0008
 27400 | -0.7663
 27500 |  -0.898
 27600 |  0.4929
 27700 | -0.2225
 27800 |  0.1035
 27900 | -0.1539
 28000 |  -0.424
 28100 |  0.1362
 28200 |  0.7535
 28300 |  0.4159
 28400 | -0.9926
 28500 |  0.6703
 28600 | -0.8153
 28700 | -0.1003
 28800 |  0.1584
 28900 |  0.1949
 29000 |  0.0095
 29100 |  0.3315
 29200 | -0.8626
 29300 |   0.231
 29400 |  0.9637
 29500 | -0.0451
 29600 | -0.7121
 29700 |  0.4119
 29800 | -0.2828
 29900 | -0.8954
 30000 |  0.4259
(95 rows)

--float
select a.id,case when a.float_x = b.float_x then a.float_x else 0 end from 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as a
 inner join 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as b
 on a.float_x=b.float_x and a.id=b.id order by a.id;
  id   |       case       
-------+------------------
 20100 |  0.2385298758745
 20200 |  0.0580289084464
 20300 |  0.6849277922884
 20400 | -0.0661425888538
 20500 | -0.9411893058568
 20600 |  -0.421802775003
 20700 | -0.0062424736097
 20800 | -0.4381286678836
 20900 |  0.0195703785866
 21000 |  0.7032254487276
 21100 |  0.7956871092319
 21200 | -0.1537507371977
 21300 | -0.9092803876847
 21500 |  0.7278142031282
 21600 |  0.6556930961087
 21700 | -0.6854076851159
 21800 | -0.9699390353635
 21900 | -0.7610640525818
 22000 |  0.0676438771188
 22100 | -0.4817214775831
 22200 | -0.4696142822504
 22300 | -0.0853876248002
 22400 | -0.5091033829376
 22500 | -0.1757164308801
 22600 | -0.9978852709755
 22700 | -0.9884481290355
 22800 | -0.0677609927952
 22900 | -0.0725406128913
 23000 |  0.9792706528679
 23100 | -0.0424462985247
 23200 | -0.9274556813762
 23300 | -0.1771280961111
 23400 |  0.8797101369128
 23500 | -0.4368732012808
 23700 | -0.8118678648025
 23800 |   0.031383481808
 23900 | -0.8899431545287
 24000 | -0.7690244494006
 24100 |  -0.117745497264
 24200 |  0.2506369184703
 24300 | -0.8803994320333
 24400 | -0.8969079814851
 24500 | -0.8247731830925
 24600 |  0.7572206100449
 24700 |  0.3381459955126
 24800 | -0.0593837508932
 24900 |  0.2972988728434
 25000 |  0.3159374408424
 25100 |  0.4349688040093
 25200 | -0.7376694818959
 25300 |  0.8019705852494
 25400 |   0.486448709853
 25500 | -0.5791148077697
 25600 | -0.6608867188916
 25700 |  0.6700058346614
 25800 |  0.5864146631211
 25900 | -0.2577141970396
 26000 | -0.5054014408961
 26100 | -0.7056470653042
 26200 | -0.5365761313587
 26300 | -0.5312578706071
 26400 |  0.3738860134035
 26500 |  0.3319177655503
 26600 |  0.0568039854988
 26700 |  0.1057071257383
 26800 |  0.6168211456388
 26900 | -0.1966684162617
 27000 | -0.9483382562175
 27100 |  0.1628344571218
 27200 |   0.106381370686
 27300 | -0.5207121968269
 27400 |  0.7238272819668
 27500 | -0.3108511008322
 27600 |  0.7435929039493
 27700 |  -0.266979470849
 27800 |  0.3859777897596
 27900 |  -0.104260857217
 28000 |  0.6134519930929
 28100 |  0.7697814563289
 28200 |  0.5688858507201
 28300 | -0.3865162627772
 28400 | -0.9180630240589
 28500 | -0.8680690228939
 28600 | -0.0178541680798
 28700 | -0.7470493698493
 28800 |  0.3852444440126
 28900 |  0.3435527011752
 29000 |  0.8663844373077
 29100 |  0.0912945484743
 29200 |  0.1385649600998
 29300 |   0.859218031168
 29400 |  0.6947398427874
 29500 | -0.6014152960852
 29600 |   0.125521114096
 29700 | -0.9622763320804
 29800 |  0.9119450561702
 29900 | -0.6377154625952
 30000 |  0.4007705943659
(98 rows)

--numeric
select a.id,case when a.nume_x = b.nume_x then a.nume_x else 0 end from 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as a
 inner join 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as b
 on a.nume_x=b.nume_x and a.id=b.id order by a.id;
  id   |         case         
-------+----------------------
 20100 |   -0.458560611121356
 20200 |   -0.506388329900801
 20300 |    0.129737394861877
 20400 |  -0.0642513819038868
 20500 |    0.493912846781313
 20600 |    0.811955966986716
 20800 |    0.552436067722738
 20900 |    0.801311953924596
 21000 |   0.0573885757476091
 21100 |    0.468227957375348
 21300 |   -0.305957793258131
 21400 |    -0.26552343275398
 21500 |    0.672722100280225
 21700 |   -0.371823594905436
 21800 |    0.508550180122256
 22000 |   -0.462772773578763
 22100 |   0.0594282504171133
 22200 |   -0.784928645938635
 22300 |  -0.0332952961325645
 22400 |    0.416448422707617
 22500 |    0.676187125965953
 22600 |    0.256147186271846
 22700 |   -0.182610848918557
 22800 |   -0.104006830602884
 22900 |    0.659985916689038
 23000 |   -0.188860209658742
 23100 |  -0.0501037240028381
 23200 |   -0.818185434676707
 23300 |  -0.0466132238507271
 23400 |   -0.717739892192185
 23600 |   -0.893695692531765
 23700 |   -0.271034295670688
 23800 |   -0.873226047493517
 23900 |    0.814018675126135
 24000 |    0.833681549876928
 24100 |    0.288806070573628
 24200 |    0.913442072458565
 24400 |    0.583377392031252
 24500 |   -0.984062462113798
 24600 |   -0.803580255247653
 24700 |    0.588883420452476
 24800 |    0.472813767381012
 24900 |    0.139033701270819
 25000 |    -0.35831315163523
 25100 |   -0.224808412604034
 25200 |    0.844137607142329
 25300 |   0.0306076509878039
 25400 |  -0.0880945064127445
 25500 |    0.320389171130955
 25600 |  0.00806342903524637
 25700 |   -0.713439894840121
 25800 |    0.706211283802986
 25900 |    0.383029392920434
 26000 |   -0.841332031413913
 26100 |   -0.628900470212102
 26200 |   -0.506501241587102
 26300 |    0.566975984722376
 26400 |   -0.948467234149575
 26500 |   -0.395507866516709
 26600 |   -0.533269652165473
 26700 |    0.207269317470491
 26800 |   -0.442595122382045
 26900 |   -0.873382803052664
 27000 |    0.148534266278148
 27100 |    0.214222522452474
 27200 |   -0.718994683586061
 27300 |    -0.58093707356602
 27400 |    0.993760737590492
 27500 |    0.550148699432611
 27600 |   -0.693748875521123
 27700 |   -0.922993722371757
 27800 |   -0.635598768480122
 27900 |    0.162692498415709
 28000 |    0.887757681310177
 28100 |    0.317002320662141
 28200 |    0.726645592600107
 28300 |   -0.773391515016556
 28400 |    0.977841095067561
 28500 |    -0.70831117965281
 28600 |   -0.697572113014758
 28700 |  -0.0260046627372503
 28800 |    0.871303737163544
 28900 |   -0.798875930719078
 29000 |   -0.216976442374289
 29100 |    0.408587444573641
 29200 |    0.294308507815003
 29300 |    0.434163339436054
 29400 |    0.537739635445178
 29500 | -0.00150082446634769
 29600 |     0.39022592548281
 29700 |   -0.429571836255491
 29800 |    0.629790987819433
 29900 |     0.68250407744199
 30000 |   -0.186099912039936
(94 rows)

--small serial
select a.id,case when a.smlsrl_x = b.smlsrl_x then a.smlsrl_x else 0 end from 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as a
 inner join 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as b
 on a.smlsrl_x=b.smlsrl_x and a.id=b.id order by a.id;
  id   | case 
-------+------
 20100 |  -23
 20200 |    5
 20300 |   12
 20400 |   19
 20500 |   31
 20600 |  -18
 20700 |  -27
 20800 |    9
 20900 |    1
 21000 |  -26
 21100 |  -14
 21200 |  -19
 21300 |   -2
 21400 |    2
 21500 |    4
 21600 |   12
 21700 |   24
 21800 |   27
 21900 |   -2
 22000 |  -24
 22100 |  -29
 22200 |  -17
 22300 |   31
 22400 |  -28
 22500 |  -13
 22600 |  -22
 22700 |   29
 22800 |   -4
 22900 |  -24
 23000 |  -22
 23100 |   19
 23200 |  -28
 23300 |  -18
 23400 |   -3
 23500 |   18
 23600 |   14
 23700 |   -1
 23800 |  -21
 23900 |   17
 24000 |   14
 24100 |   31
 24200 |  -32
 24300 |   16
 24400 |  -14
 24500 |   12
 24600 |  -10
 24700 |    1
 24800 |   21
 24900 |  -26
 25000 |    5
 25100 |   -2
 25200 |   26
 25300 |   19
 25400 |    7
 25500 |   31
 25600 |   -9
 25700 |    6
 25800 |   -2
 25900 |   19
 26000 |  -32
 26100 |   -8
 26200 |    4
 26300 |   29
 26400 |    2
 26500 |  -29
 26600 |  -18
 26700 |  -33
 26800 |   -7
 26900 |  -24
 27000 |   30
 27100 |   14
 27200 |   18
 27300 |   26
 27400 |   -9
 27500 |  -11
 27600 |   19
 27700 |   12
 27800 |  -23
 27900 |    8
 28000 |  -19
 28100 |   25
 28200 |  -25
 28300 |  -16
 28400 |  -20
 28500 |   22
 28600 |   -5
 28700 |  -16
 28800 |  -12
 28900 |   11
 29000 |   25
 29100 |   -8
 29200 |   -9
 29300 |    4
 29400 |  -12
 29500 |  -19
 29600 |    2
 29700 |   14
 29800 |    2
 29900 |    0
 30000 |  -18
(100 rows)

--serial
select a.id,case when a.serial_x = b.serial_x then a.serial_x else 0 end from 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as a
 inner join 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as b
 on a.serial_x=b.serial_x and a.id=b.id order by a.id;
  id   |   case   
-------+----------
 20100 |  -529567
 20200 |   547557
 20300 |    -9050
 20400 |  1222399
 20500 | -1692638
 20600 |   702299
 20700 |   848189
 20800 |   995859
 20900 | -2010791
 21000 |  1125930
 21100 |  1864098
 21200 |   -98633
 21300 |  -886411
 21400 |  1766305
 21500 |  1169481
 21600 |  -478443
 21700 |  -876096
 21800 |  1197627
 21900 | -1558227
 22000 |  1787273
 22100 |   -17146
 22200 |  -449609
 22300 | -2131147
 22400 |  -303791
 22500 | -1030283
 22600 |  1138290
 22700 |  -692397
 22800 |  -675527
 22900 |   229740
 23000 |  -375984
 23100 |  1434337
 23200 |   271749
 23300 |  -236073
 23400 | -1758983
 23500 | -1667854
 23600 |   590955
 23700 |   430989
 23800 |  1175157
 23900 |  1075577
 24000 |   -22785
 24100 |  -809797
 24200 |   649900
 24300 |   306692
 24400 |   204043
 24500 |  1869281
 24600 |  1831309
 24700 |  2132876
 24800 |  -539649
 24900 |  -852236
 25000 |   -92908
 25100 | -1480961
 25200 |  -342549
 25300 | -1475114
 25400 | -1548032
 25500 |   699463
 25600 |   425989
 25700 | -1401672
 25800 | -1884106
 25900 |   348250
 26000 |  -538464
 26100 | -1546037
 26200 |  -348524
 26300 |  1540971
 26400 |  -820138
 26500 |  1586343
 26600 |  -807708
 26700 |  1840495
 26800 |   454568
 26900 |  1199482
 27000 |    35622
 27100 |  -482563
 27200 |   396381
 27300 |  -610701
 27400 |   971218
 27500 |  -633959
 27600 |  1106415
 27700 |  1799084
 27800 | -1095165
 27900 | -1856418
 28000 | -1869355
 28100 |   950370
 28200 |  1619512
 28300 | -1704525
 28400 | -1957400
 28500 | -1114844
 28600 |  1252425
 28700 |   297772
 28800 |  -836617
 28900 |  1317346
 29000 | -1447630
 29100 |   644963
 29200 |  1810573
 29300 |  1533624
 29400 | -1371414
 29500 | -1290132
 29600 | -2141800
 29700 |    43027
 29800 | -1981032
 29900 | -1484536
 30000 | -1353732
(100 rows)

--big serial
select a.id,case when a.bigsrl_x = b.bigsrl_x then a.bigsrl_x else 0 end from 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as a
 inner join 
(select * from strom_test where id between 20001 and 30000 and id%100=0 and id::text not like '' ) as b
 on a.bigsrl_x=b.bigsrl_x and a.id=b.id order by a.id;
  id   |       case        
-------+-------------------
 20100 |  2938945979195326
 20200 | -8608401963530322
 20300 | -7075365270428058
 20400 |  8181438259612615
 20500 | -8029133367767204
 20600 |  4092028854313419
 20700 | -2601074018849128
 20800 | -1842676184144413
 20900 | -8434412357333025
 21000 |  4748375740624404
 21100 |  1837735597364478
 21200 | -7373867679343444
 21300 |  2145267393335656
 21400 |  2467814918959661
 21500 |  1626822492330918
 21600 | -2799481759664702
 21700 | -4278868521624011
 21800 |  2513183715069264
 21900 | -1157291049714975
 22000 | -8775035879887471
 22100 |  8782688248888558
 22200 |  4295841364794409
 22300 |  3031910821195154
 22400 |  6257462055406666
 22500 | -7722071466988012
 22600 | -9184283307565646
 22700 | -6118042031089517
 22800 |  4141649083930510
 22900 | -1843121829951046
 23000 | -5652511221660778
 23100 | -7456921798408405
 23200 | -2729892663720411
 23300 |  7886382922729521
 23400 | -3063683348573454
 23500 |   586674562490434
 23600 |  2464919329318109
 23700 | -4726759719979450
 23800 |  3394604179228984
 23900 | -2077776474814808
 24000 | -6550538383028912
 24100 | -7740478485637890
 24200 |  1955691350821175
 24300 |  5543089628350775
 24400 | -8973114995575357
 24500 |  -315098738072748
 24600 |  5194973337779962
 24700 |  2901960340772225
 24800 |  -875725071237251
 24900 | -1611705745047290
 25000 | -4108557159778746
 25100 | -1292484929292075
 25200 | -8848340191262278
 25300 |  3911432712241545
 25400 |   717061394410766
 25500 | -3517920807785332
 25600 |  7644502845793763
 25700 |  5742925368482857
 25800 |  4419920478894818
 25900 | -1119157040116138
 26000 | -7586892310558802
 26100 |  9198093689006588
 26200 |  1552684716782518
 26300 | -2623611370108616
 26400 |  6445876000029934
 26500 |  6027143867222983
 26600 |  2212474320028959
 26700 | -8174401840461382
 26800 |  7009596668152644
 26900 | -7249245372139176
 27000 | -3624280106593157
 27100 |  4492092791509222
 27200 | -1774742763732992
 27300 | -2261140740488823
 27400 | -8339830468736188
 27500 | -4323531568707535
 27600 |  3280656918749643
 27700 | -2477146293755970
 27800 | -8383854278657180
 27900 | -3700228279576298
 28000 |  1884020449868776
 28100 |  7421090097438654
 28200 |  1507187518330634
 28300 |  7448514545955373
 28400 |  3524309700487152
 28500 | -8262376780866781
 28600 | -5578090631105872
 28700 |  -406101011946537
 28800 |  1320775037057237
 28900 | -8025849667700916
 29000 |  3261043186088280
 29100 |  9150402990627094
 29200 | -6233383807320326
 29300 | -6885184582417646
 29400 | -6285406796261622
 29500 | -2770808386197914
 29600 | -4444192533085422
 29700 | -5002994939668201
 29800 |  6822134831879553
 29900 |   178916388952343
 30000 |  2731118516106166
(100 rows)

